### PR TITLE
Port drawing example to 0.5

### PR DESCRIPTION
**Pull request by euclio**
cc #100

### CODE REVIEW

File: examples/drawing.rs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        let widgets = view_output!();
+
+        model.handler.init(&widgets.area);
+
+        sender.command(|out, shutdown| {
+            shutdown
+                .register(async move {
+                    loop {
+                        tokio::time::sleep(Duration::from_millis(20)).await;
+                        out.send(AppMsg::UpdatePoints);
```
Comment (AaronErhardt) [inline examples/drawing.rs:136]: You're normally not supposed to do so, but you could move a clone of the sender into the closure to send messages directly to input. Or even better, remove the AppMsg::UpdatePoints variant and update the points only from `update_cmd`.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Looks good, just one minor thing to improve.

File: examples/drawing.rs (hunk 1 above)
Comment (euclio) [reply examples/drawing.rs:136]: > You're normally not supposed to do so, but you could move a clone of the sender into the closure to send messages directly to input.

I'm a bit wary of doing something you're "not supposed to do" in exemplary code. Maybe it's fine with a comment?

> remove the AppMsg::UpdatePoints variant and update the points only from `update_cmd`.

I tried this, but then a view update isn't triggered.

Comment (AaronErhardt) [reply examples/drawing.rs:136]: > I tried this, but then a view update isn't triggered.

Actually, `update_cmd` should trigger a view update.

Comment (euclio) [reply examples/drawing.rs]: You're right, I was missing a call to `draw`. Fixed, and simplified the reset logic a bit, too.

**Review by AaronErhardt — APPROVED** [review]
Thanks!

**Pull request merged**